contracts/axelar-gateway/src/contract.rs

```diff
@@ -93,13 +94,17 @@ impl AxelarGatewayInterface for AxelarGateway {
             )
     }
 
-    fn is_message_executed(env: Env, source_chain: String, message_id: String) -> bool {
+    /// Return true if a contract call with the given payload BytesN<32> and source caller info has been executed.
+    pub fn is_message_executed(env: Env, source_chain: String, message_id: String) -> bool {
         let message_approval = Self::message_approval(&env, source_chain, message_id);
 
         message_approval == MessageApprovalValue::Executed
     }
 
-    fn validate_message(
+    /// Validate if a contract call with the given payload BytesN<32> and source caller info is approved,
+    /// preventing re-validation (i.e distinct contract calls can be validated at most once).
+    /// `caller` must be the intended `destination_address` of the contract call for validation to succeed.
```

Comment:
See EVM method doc for reference

Reply:
updated docstring to better match evm reference